src/main/java/com/dougnoel/sentinel/steps/DownloadVerificationSteps.java

```diff
@@ -65,4 +70,28 @@ public static void verifyFileWithFilenameDownloaded(String expectedFilename) thr
 	public static void clearDownloadsFolder() throws IOException {
 		DownloadManager.clearDownloadDirectory();
 	}
+
+	/**
+	 * Looks inside the most recently downloaded zip file and asserts that it does or does not contain a file with the given extension in the top-level directory.
+	 * @param assertion String " does not" for a negative check, otherwise positive check
+	 * @param expectedFileType String the file extension to check for.
+	 */
+	@Then("^I verify the most recently downloaded zip file( does not)? contains? a file with the extension (.*?)$")
+	public static void verifyFileContentsOfZip(String assertion, String expectedFileType) throws IOException {
+		File mostRecentFile = DownloadManager.getMostRecentDownloadPath().toFile();
+		List<String> fileContent = null;
+		boolean result = false;
+		boolean negate = !StringUtils.isEmpty(assertion);
+		try(var zip = new ZipFile(mostRecentFile)){
+			fileContent = zip.stream().map(ZipEntry::getName).collect(Collectors.toList());
+			result = fileContent.stream().anyMatch(fileName -> StringUtils.endsWith(fileName, expectedFileType));
+		}
+		catch(IOException ioe){
+			String message = SentinelStringUtils.format("Unable to open most recently downloaded file as zip file. Most recently downloaded file path {}", mostRecentFile.getPath());
+			throw new IOException(message, ioe);
+		}
+		assertEquals(SentinelStringUtils.format("Expected zip file to {}contain a file with extension {}. Zip file location: {} Files in zip: {}",
+						(negate ? "not ": ""), expectedFileType, mostRecentFile, StringUtils.join(fileContent, ", ")),
```

Comment:
This error message needs to be very clear if we aren't searching subdirectories in the zip.

Reply:
The file enumeration actually does find subdirectories (except nested zips) but that is clear in the method description.